src/wrappers/nats.js

```diff
@@ -0,0 +1,198 @@
+    };
+};
+
+/**
+ * Handle nats subscribe callback event.
+ *
+ * @param {String} callback_msg received message.
+ * @param {String} callback_reply received reply.
+ * @param {String} callback_subject received subject.
+ * @param {Number} callback_sid received subscribe id.
+ * @param {Function} callback  callback function.
+ * @param {String} serverHostname nats server host name.
+ * @param {Boolean} isRequestCall true if this subscribe call came from nats Client.request.
+ */
+function natsSubscribeCallbackMiddleware(
+    callback_msg,
+    callback_reply,
+    callback_subject,
+    callback_sid,
+    callback,
+    serverHostname,
+    isRequestCall
+) {
+    let originalHandlerSyncErr;
+    try {
+        // Initialize tracer and evnets.
+        tracer.restart();
+        const { slsEvent: natsEvent, startTime: natsStartTime } =
+        eventInterface.initializeEvent(
+            'nats',
+            serverHostname,
+            isRequestCall ? 'requestMessageListener' : 'subscribeMessageListener',
+            'trigger'
+        );
+        tracer.addEvent(natsEvent);
+
+        // Getting message data.
+        const triggerMetadata = {};
+        if (callback_msg) triggerMetadata.msg = callback_msg;
+        if (callback_reply) triggerMetadata.reply = callback_reply;
+        if (callback_subject) triggerMetadata.subject = callback_subject;
+        if (callback_sid) triggerMetadata.sid = callback_sid;
+        // Finalize nats event.
+        eventInterface.finalizeEvent(natsEvent, natsStartTime, null, triggerMetadata);
+        const { slsEvent: nodeEvent, startTime: nodeStartTime } = eventInterface.initializeEvent(
+            'node_function', 'messageHandler', 'messageReceived', 'runner'
+        );
+        let runnerResult;
+        try {
+            runnerResult = callback(callback_msg, callback_reply, callback_subject, callback_sid);
+        } catch (err) {
+            originalHandlerSyncErr = err;
+        }
+        const originalHandlerName = callback.name;
+        if (originalHandlerName) {
+            nodeEvent.getResource().setName(originalHandlerName);
+        }
+        // Handle and finalize async user function.
+        if (utils.isPromise(runnerResult)) {
+            let originalHandlerAsyncError;
+            runnerResult.catch((err) => {
+                originalHandlerAsyncError = err;
+                throw err;
+            }).finally(() => {
+                eventInterface.finalizeEvent(nodeEvent, nodeStartTime, originalHandlerAsyncError);
+                tracer.sendTrace(() => {});
+            });
+        } else {
+            // Finalize sync user function.
+            eventInterface.finalizeEvent(nodeEvent, nodeStartTime, originalHandlerSyncErr);
+            tracer.sendTrace(() => {});
+        }
+        tracer.addRunner(nodeEvent, runnerResult);
+    } catch (err) {
+        tracer.addException(err);
+    }
+    // Throwing error in case of sync user function.
```

Comment:
where is `return runnerResult`?

Reply:
this function doesn't return anything, but you right, its always better to return.